go/tools/go-upgrade/go_upgrade.go

```diff
@@ -0,0 +1,453 @@
+// 		  avoid permission issues. The rewrite of workflow files can
+// 		  be disabled using the no-workflow-update CLI flag.
+//
+// 		  Moreover, this command automatically bumps the bootstrap
+// 		  version of our codebase. If we are on the main branch, we
+// 		  want to use the CLI flag `main` to remember to increment
+// 		  the bootstrap version by 1 instead of 0.1.
+//
+//
+// 		- Get Current Go Version: `./go/tools/go-upgrade/go_upgrade.go get_go_version`
+//
+// 		  This command outputs the currently used Golang version of
+// 		  our codebase.
+//
+//
+// 		- Get Current bootstrap Version: `./go/tools/go-upgrade/go_upgrade.go get_bootstrap_version`
+//
+// 		  This command outputs the currently used bootstrap version
+// 		  of our codebase.
+//
+//
+// 		- Update Workflows: `./go/tools/go-upgrade/go_upgrade.go update_workflows`
+//
+// 		  This command is used after the auto Pull Request has been
+// 		  created. One of the reviewer must go and use this command
+// 		  to update the workflows files with the newer Golang version.
+// 		  It takes two arguments `go-from` and `go-to` to identify the
+// 		  old Golang version (go-from) and the new one (go-to).
+
+const (
+	goDevAPI = "https://go.dev/dl/?mode=json"
+)
+
+type latestGolangRelease struct {
+	Version string `json:"version"`
+	Stable  bool   `json:"stable"`
+}
+
+func main() {
+	noWorkflowUpdate := flag.Bool("no-workflow-update", false, "Whether or not the workflow files should be updated. Useful when using this script to auto-create PRs.")
+	allowMajorUpgrade := flag.Bool("allow-major-upgrade", false, "Defines if Golang major version upgrade are allowed.")
+	isMainBranch := flag.Bool("main", false, "Defines if the current branch is the main branch.")
+	goFrom := flag.String("go-from", "", "The original Golang version we start with.")
+	goTo := flag.String("go-to", "", "The Golang version we want to upgrade to.")
+	flag.Parse()
+
+	switch {
+	case slices.Contains(os.Args, "get_go_version"):
+		currentVersion, err := currentGolangVersion()
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println(currentVersion.String())
+	case slices.Contains(os.Args, "get_bootstrap_version"):
+		currentBootstrapVersionF, err := currentBootstrapVersion()
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println(currentBootstrapVersionF)
+	case slices.Contains(os.Args, "update_workflows"):
+		if *noWorkflowUpdate {
+			break
+		}
+		err := updateWorkflowFilesOnly(*goFrom, *goTo)
+		if err != nil {
+			log.Fatal(err)
+		}
+	default:
+		err := upgradePath(*allowMajorUpgrade, *noWorkflowUpdate, *isMainBranch)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+func updateWorkflowFilesOnly(goFrom, goTo string) error {
+	oldV, err := version.NewVersion(goFrom)
+	if err != nil {
+		return err
+	}
+	newV, err := version.NewVersion(goTo)
+	if err != nil {
+		return err
+	}
+	filesToChange, err := getListOfFilesInPaths([]string{"./.github/workflows"})
```

Comment:
What is "." here? I'm assuming it has to be the root of the repo branch, which means it can only be run from there. Which is fine, although we could try to determine that ourselves or make it an argument. Current way is fine though, I'd add a comment.

Reply:
Added a note in the usage of the go-upgrade command via c9579a436c. I think to leave it as it is right now, we can modify it later if needed.